docs/tutorials/sparse/train_gluon.md

```diff
@@ -0,0 +1,469 @@
+
+
+```python
+def save_as_libsvm(filepath, data, label):
+    with open(filepath, 'w') as openfile:
+        for row_idx in range(data.shape[0]):
+            data_sample = data[row_idx]
+            label_sample = label[row_idx]
+            col_idxs = data_sample.indices.asnumpy().tolist()
+            values = data_sample.data.asnumpy().tolist()
+            label_str = str(label_sample.asscalar())
+            value_strs = ['{}:{}'.format(idx, value) for idx, value in zip(col_idxs, values)]
+            value_str = " ".join(value_strs)
+            sample_str = '{} {}\n'.format(label_str, value_str)
+            openfile.write(sample_str)
+```
+
+
+```python
+filepath = 'dataset.libsvm'
+save_as_libsvm(filepath, data, label)
+```
+
+We have now written the `data` and `label` to disk, and can inspect the first 10 lines of the file:
+
+
+```python
+with open(filepath, 'r') as openfile:
+    lines = [openfile.readline() for _ in range(10)]
+for line in lines:
+    print(line[:80] + '...' if len(line) > 80 else line)
+```
+
+    0.0 35454:0.22486156225204468 80954:0.39130592346191406 81941:0.1988530308008194...
+    1.0 37029:0.5980494618415833 52916:0.15797750651836395 71623:0.32251599431037903...
+    1.0 89962:0.47770974040031433 216426:0.21326342225074768 271027:0.18589609861373...
+    1.0 7071:0.9432336688041687 81664:0.7788773775100708 117459:0.8166475296020508 4...
+    0.0 380966:0.16906292736530304 394363:0.7987179756164551 458442:0.56873309612274...
+    0.0 89361:0.9099966287612915 141813:0.5927085280418396 282489:0.293381005525589 ...
+    0.0 150427:0.4747847020626068 169376:0.2603490948677063 179377:0.237988427281379...
+    0.0 49774:0.2822582423686981 91245:0.5794865489006042 102970:0.7004560232162476 ...
+    1.0 97133:0.0024336236529052258 109855:0.9895315766334534 116765:0.2465638816356...
+    0.0 803440:0.4020800292491913
+    
+
+
+Some storage overhead is introduced by serializing the data as characters (with spaces and colons). `dataset.libsvm` is 250 KBs but the original `data` and `label` were 132 KBs combined. Compared with the 4GB dense `NDArray` though, this isn't a huge issue.
+
+### Reading Sparse Data
+
+Using [`LibSVMIter`](https://mxnet.incubator.apache.org/api/python/io/io.html?highlight=libsvmiter), we can quickly and easily load data into batches ready for training. Although Gluon [`Dataset`](https://mxnet.incubator.apache.org/versions/master/api/python/gluon/data.html?highlight=dataset#mxnet.gluon.data.Dataset)s can be written to return sparse arrays, Gluon [`DataLoader`](https://mxnet.incubator.apache.org/versions/master/api/python/gluon/data.html?highlight=dataloader#mxnet.gluon.data.DataLoader)s currently convert each sample to dense before stacking up to create the batch. As a result, [`LibSVMIter`](https://mxnet.incubator.apache.org/api/python/io/io.html?highlight=libsvmiter) is the recommended method of loading sparse data in batches.
```

Comment:
Have you tested this style of link on beta? I believe I saw that it doesn't parse. Could be a bug on beta...

Reply:
Works on main website, it doesn't work on beta, but this should be fixed rather than worked around.